### PR TITLE
W-16734896 Add AsyncAPI design docs 10/24 cc

### PR DESCRIPTION
**Pull request by chris-campbell9**

### CODE REVIEW

File: modules/ROOT/pages/des-create-api-specs.adoc [hunk 1]
```diff
@@ -58,6 +58,12 @@ include::partial$acb-reusable-steps.adoc[tags="no-project-name-reuse"]
 See xref:start-add-folders.adoc[].
 
 include::partial$acb-reusable-steps.adoc[tags="proj-directory-warn"]
+| *API Type* | The type of API spec to create.
+
+Available options are *REST API* and *AsyncAPI*.
+
+Select *REST API* to use the <<example-oas-api-spec,example>> in this procedure.
+
 | *API Specification Language* | See xref:des-designing-api-specs.adoc#supported-spec-languages[Supported OAS and RAML Versions].
```
Comment (carobazan) [inline modules/ROOT/pages/des-create-api-specs.adoc:67]: Should include Async : [Supported OAS, RAML and Async Versions]

**Review by lbauret — APPROVED** [review]
LGTM!

File: modules/ROOT/pages/des-create-asyncapi-api-specs.adoc [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+:page-deployment-options: cloud-ide, desktop-ide
+
+include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]
+
+Use Anypoint Code Builder to create, import, and test AsyncAPI API specs before publishing to Anypoint Exchange:
+
+. Create an AsyncAPI API spec project in Anypoint Code Builder from the IDE or by importing the spec from Design Center:
+
+* <<design-new-api-spec, Create an AsyncAPI API spec project from the IDE>>
+* <<import-spec, Import an AsyncAPI API spec from Design Center>>
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/des-create-asyncapi-api-specs.adoc:11]:
```suggestion
** <<design-new-api-spec, Create an AsyncAPI API spec project from the IDE>>
** <<import-spec, Import an AsyncAPI API spec from Design Center>>
```
I think this is necessary to render the bulleted list items nested under the first ordered list item. If you tested in a local build and this rendered properly, ignore this comment.

File: modules/ROOT/pages/des-create-asyncapi-api-specs.adoc [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+.. Display the console menu.
+. Click channels in the console or select items from the menu to view different parts of your spec.
+For example, click *PUBLISH*:
++
+image::des-asyncapi-api-console-detail.png["API Console with Publish details"]
+
+[[example-asyncapi-api-spec]]
+== Example AsyncAPI (YAML) API Specification
+
+If you created an AsyncAPI 2.6 (YAML) project, you can replace the initial spec with this example code:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/des-create-asyncapi-api-specs.adoc:141]:
```suggestion
If you created an AsyncAPI 2.6 (YAML) project, you can replace the initial spec with this example code:
```
What's the purpose of sharing this example? It works as a template for users to continue building? Or just as an example so they have an idea of what a spec looks like? I think I'm a bit confused by the sentence before the example, could you make it more intentional? Maybe something like these options:

- To see an example specification, create an AsyncAPI 2.6 (YAML) project and replace the initial spec with the following code:
- Create an AsyncAPI 2.6 (YAML) project, replace the initial spec with the following code, and use it as a starting point to create your spec:

**Review by Cristian-Venticinque — APPROVED** [review]
Approved with two minor comments

File: modules/ROOT/pages/des-create-asyncapi-api-specs.adoc (hunk 3 above)
Comment (chris-campbell9) [reply modules/ROOT/pages/des-create-asyncapi-api-specs.adoc:141]: Hi @Cristian-Venticinque, you raise good points! I thought about this a while, but I think the current text makes sense in the context of the rest of the procedure. 

Earlier in the procedure the customer creates an AsyncAPI project, which generates boilerplate code. If they'd selected the option to create an AsyncAPI 2.6 (YAML) project, the boilerplate code reflects this. By the time the customer gets to the end of the procedure, they have the option to replace this boilerplate code with the example in this section. It's entirely optional to do so, and it only applies if they created an AsyncAPI 2.6 (YAML) project. 

I'm thinking the suggested changes change the meaning of this.

File: modules/ROOT/pages/des-create-asyncapi-api-specs.adoc (hunk 2 above)
Comment (chris-campbell9) [reply modules/ROOT/pages/des-create-asyncapi-api-specs.adoc:11]: Hi @Cristian-Venticinque, the local build seems to correctly render the original syntax.

**Pull request merged**